src/litserve/server.py

```diff
@@ -625,8 +657,21 @@ async def predict(request: self.request_type, background_tasks: BackgroundTasks)
 
             self.request_queue.put_nowait((response_queue_id, uid, time.monotonic(), payload))
 
-            await event.wait()
-            response, status = self.response_buffer.pop(uid)
+            async def wait_for_response():
```

Comment:
@aniketmaurya , This approach for non streaming case seems to impact the benchmark a lot. 
In this portion, it is just checking if the client is disconnected and the actual task is yet to be cancelled from running loop.

Reply:
Seems to got improved with abit of refactoring the approach 
<img width="546" alt="image" src="https://github.com/user-attachments/assets/0df0f0e7-2de6-4177-8597-902e4bc496b9">